lib/boom_notifier/mail_notifier.ex

```diff
@@ -31,19 +31,32 @@ defmodule BoomNotifier.MailNotifier do
           {:mailer, module()} | {:from, String.t()} | {:to, String.t()} | {:subject, String.t()}
   @type options :: [option]
 
+  @impl BoomNotifier.Notifier
+  def validate!(options) do
+    with _mailer <- Keyword.fetch!(options, :mailer),
+         _from <- Keyword.fetch!(options, :from),
+         _to <- Keyword.fetch!(options, :to),
+         _subject <- Keyword.fetch!(options, :subject) do
+      nil
+    end
+  end
+
   @impl BoomNotifier.Notifier
   @spec notify(list(%ErrorInfo{}), options) :: no_return()
   def notify(error_info, options) do
-    [mailer: mailer, from: email_from, to: email_to, subject: subject] = options
-
-    email =
-      new_email()
-      |> to(email_to)
-      |> from(email_from)
-      |> subject("#{subject}: #{hd(error_info) |> Map.get(:reason)}")
-      |> html_body(HTMLContent.build(error_info))
-      |> text_body(TextContent.build(error_info))
-
-    mailer.deliver_later(email)
+    with {:ok, mailer} <- Keyword.fetch(options, :mailer),
+         {:ok, email_from} <- Keyword.fetch(options, :from),
+         {:ok, email_to} <- Keyword.fetch(options, :to),
+         {:ok, subject} <- Keyword.fetch(options, :subject) do
+      email =
+        new_email()
+        |> to(email_to)
```

Comment:
Here we can assume that keywords are actually present, so no need to double-check, right?

We can get rid of the `with` block and have, for instance:

```ex
email =
  new_email()
  |> to(options[:to])
  |> from(options[:from])
  ....
```